File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int>ans;
        if(!root) return ans;
        queue<TreeNode*>q;
        q.push(root);

        while(!q.empty()){
            int sz=q.size();
           int maxm=INT_MIN;
            while(sz--){
                TreeNode* node=q.front();
                q.pop();
               
                if(node->val>maxm){
                    maxm=node->val;
                }
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
            }
ans.push_back(maxm);

        }
        return ans;
    }
};

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var largestValues = function(root) {
     let ans=[];
        if(!root) return ans;
        let q=[];
        q.push(root);

        while(q.length>0){
            let sz=q.length;
           let maxm=-Infinity;
            while(sz--){
                const node=q.shift();
               
               
                if(node.val>maxm){
                    maxm=node.val;
                }
                if(node.left) q.push(node.left);
                if(node.right) q.push(node.right);
            }
ans.push(maxm);

        }
        return ans;
};